app/Policies/ThreadPolicy.php

```diff
@@ -41,4 +43,9 @@ public function lock(User $user): bool
     {
         return $user->isAdmin() || $user->isModerator();
     }
+
+    public function markAsSpam(User $user, Thread $thread): bool
+    {
+        return $thread->spammers->doesntContain($user);
```

Comment:
We'll also need to filter out moderator and admin replies which shouldn't be reportable as spam.

Reply:
noted, should we do this in the view instead? Maybe we can do something like `$reply->is_admin_or_moderator_reply ? 'dont show' : 'show mark as spam button'`

Reply:
you can review here https://github.com/laravelio/laravel.io/pull/886/commits/b62eccab14d1a40e4d0eb36444c59f242a9957f3

Reply:
In the policy is fine. That should also hide the button when checking the policy.